mos_tests/environment/networking.py

```diff
@@ -0,0 +1,254 @@
+
+
+logger = logging.getLogger(__name__)
+
+
+class NotFound(Exception):
+    message = "Not Found."
+
+
+class NetworkCheckBase(object):
+    """Class contains common methods for networks checks"""
+
+    @pytest.fixture(autouse=True)
+    def init(self, env, os_conn):
+        self.env = env
+        self.os_conn = os_conn
+        self.cirros_creds = {'username': 'cirros',
+                             'password': 'cubswin:)'}
+
+    def run_on_vm(self, vm, vm_keypair=None, command='uname',
+                  vm_login="cirros", timeout=3 * 60, vm_password='cubswin:)'):
+        """Execute command on vm and return dict with results
+
+        :param vm: server to execute command on
+        :param vm_keypair: keypair used during vm creating
+        :param command: command to execute
+        :param vm_login: username to login to vm via ssh
+        :param vm_password: password to login to vm via ssh
+        :param timeout: type - int or None
+            - if None - execute command and return results
+            - if int - wait `timeout` seconds until command exit_code will be 0
+        :returns: Dictionary with `exit_code`, `stdout`, `stderr` keys.
+            `Stdout` and `stderr` are list of strings
+        """
+        results = []
+
+        def execute():
+            with self.os_conn.ssh_to_instance(self.env, vm, vm_keypair,
+                                              username=vm_login,
+                                              password=vm_password) as remote:
+                result = remote.execute(command)
+                results.append(result)
+                return result
+
+        logger.info('Executing `{cmd}` on {vm_name}'.format(
+            cmd=command,
+            vm_name=vm.name))
+
+        if timeout is None:
+            execute()
+        else:
+            err_msg = "SSH command: `{command}` completed with 0 exit code"
+            wait(lambda: execute()['exit_code'] == 0,
+                 sleep_seconds=(1, 60, 5), timeout_seconds=timeout,
+                 expected_exceptions=(Exception,),
+                 waiting_for=err_msg.format(command=command))
+        return results[-1]
+
+    def check_ping_from_vm(self, vm, vm_keypair=None, ip_to_ping=None,
+                           timeout=3 * 60, vm_login='cirros',
+                           vm_password='cubswin:)'):
+        logger.info('Expecting that ping from VM should pass')
+        # Get ping results
+        result = self.check_ping_from_vm_helper(
+            vm, vm_keypair, ip_to_ping,
+            timeout, vm_login, vm_password)
+
+        error_msg = (
+            'Instance has NO connection, but it should have.\n'
+            'EXIT CODE: "{exit_code}"\n'
+            'STDOUT: "{stdout}"\n'
+            'STDERR {stderr}').format(**result)
+
+        # As ping should pass we expect '0' in exit_code
+        assert 0 == result['exit_code'], error_msg
+
+    def check_no_ping_from_vm(self, vm, vm_keypair=None, ip_to_ping=None,
```

Comment:
This (and some other) methods are not need in NFV tests. Let's leave here only necessary methods.